### PR TITLE
feature unit tests

### PR DESCRIPTION
**Pull request by ben-vr**

### CODE REVIEW

File: tests/lib/amazon.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        contentType: 'image/png',
+        bucket: 'bucket',
+        key: 'key',
+        encryption: 'AE-256',
+      });
+
+      expect(s3PromiseMock).toBeCalledTimes(1);
+    });
+
+    it('Should throw a custom error', () => {
```
Comment (horstenwillem) [inline tests/lib/amazon.test.ts]: Nice test 😉

File: tests/lib/middleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// import * as Multer from 'multer';
```
Comment (horstenwillem) [inline tests/lib/middleware.test.ts]: Remove commented line

File: tests/lib/amazon.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import * as path from 'path';
+import { createClient, uploadFile, getPresignedUrl } from '../../src/lib/amazon';
+import { BadRequestError } from 'tree-house-errors';
+import { errors } from '../../src/config/error-config';
+describe('amazon', () => {
```
Comment (horstenwillem) [inline tests/lib/amazon.test.ts]: Add newline above describe block

File: tests/lib/amazon.test.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          path: path.resolve(__dirname, '../_helpers/unit_tests.png'),
+          contentType: 'image/png',
+          bucket: 'bucket',
+          key: 'key',
+          encryption: 'AE-256',
+        });
+      } catch (error) {
+        expect(error).toBeInstanceOf(BadRequestError);
+        expect(error.code).toEqual(errors.FILE_UPLOAD_ERROR.code);
+        expect(error.message).toEqual('random error occured');
```
Comment (knor-el-snor) [inline tests/lib/amazon.test.ts]: I would copy the `validateError` lib function from ess-api

File: tests/lib/amazon.test.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      }
+    });
+  });
+
+  describe('getPresignedUrl', () => {
+    it('Should succesfully return the pre-signed url', async () => {
+      const resourceUrl = await getPresignedUrl(awsClient, { bucket: 'bucket', key: 'key' });
+
```
Comment (knor-el-snor) [inline tests/lib/amazon.test.ts]: Remove newline

File: tests/lib/local.test.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      createIfNotExists(folderPath);
+
+      expect(fs.existsSync(folderPath)).toBeTruthy();
+      fs.rmdirSync(folderPath);
+    });
+  });
+
+  describe('readFile', () => {
+    it('Should return readStream when file is found', async () => {
+      const result = await readFile(path.resolve(__dirname, '../_helpers/unit_tests.png'));
```
Comment (knor-el-snor) [inline tests/lib/local.test.ts]: I would create a file in `beforeAll` and remove it in `afterAll`. This way we do not need to include files for testing

File: tests/lib/middleware.test.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        allowedFileTypes: ['application/xml'],
+      });
+
+      expect(middleware).toHaveProperty('single');
+      expect(middleware).toHaveProperty('fields');
+      expect(middleware).toHaveProperty('array');
+    });
+
+    // Mock data
+    const request = httpMocks.createRequest({
```
Comment (knor-el-snor) [inline tests/lib/middleware.test.ts]: Move these declarations to top of describe

File: tests/lib/middleware.test.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      const options: MultipartOptions = {
+        fileSize: 100,
+        destination: '/',
+        allowedFileTypes: ['application/xml'],
+        validator: {
+          schema: { body: { name: Joi.string().required() } },
+        },
+      };
+
+      const callbackFunction = (error: ApiError) => {
```
Comment (knor-el-snor) [inline tests/lib/middleware.test.ts]: expect.assertions

File: tests/lib/middleware.test.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    it('Should throw an error when the request body is invalid', () => {
+      const options: MultipartOptions = {
+        fileSize: 100,
+        destination: '/',
+        allowedFileTypes: ['application/xml'],
+        validator: {
+          schema: { body: { anotherKey: Joi.string().required() } },
+        },
+      };
+
```
Comment (knor-el-snor) [inline tests/lib/middleware.test.ts]: expect.assertions

**Pull request merged**